open-apparel-registry/open-apparel-registry
# Make period_start_date editable

## Overview

The field `period_start_date` on ApiLimits was set to `auto_add_now`.
This makes the field read-only, and hides it in the Django admin. In
order to make the field editable in the admin, it has been updated to
use `timezome.now` as a default and forbid null values instead of using
`auto_add_now`.

Connects #1427 

## Demo

<img width="598" alt="Screen Shot 2021-07-20 at 10 27 13 AM" src="https://user-images.githubusercontent.com/21046714/126341997-1188fece-7f5e-4e99-a388-17be2b4b8131.png">
<img width="1075" alt="Screen Shot 2021-07-20 at 10 27 37 AM" src="https://user-images.githubusercontent.com/21046714/126342005-5f731994-ec26-4521-a8f1-1421032eb6fa.png">

## Testing Instructions

* Run `./scripts/update` and `./scripts/server`
* Navigate to the [Django admin](http://localhost:8081/admin/) and add a new API limit. Ensure the `period_start_date` is set appropriately. 
* Update the ApiLimit.period_start_date and confirm it's updated as expected. 
* View the history for the limit and confirm that the period_start_date changes were tracked. 

## Checklist
